File: exam-activities/instituto-app/src/data/entities/Matricula.ts
import { BaseEntity, Column, Entity, JoinTable, ManyToMany, ManyToOne, OneToMany, PrimaryColumn, PrimaryGeneratedColumn } from "typeorm";
import { Alumno } from "./Alumno";
import { Asignatura } from "./Asignatura";

@Entity('matriculas')
export class Matricula extends BaseEntity
{
    @PrimaryGeneratedColumn()
    id: number;

    @Column('integer')
    year: number;

    @ManyToOne(() => Alumno, (alumno) => alumno.matriculas)
    alumno: Alumno;

    @ManyToMany(() => Asignatura, (asignatura) => asignatura.matriculas)
    @JoinTable(
    {
      name: 'asignaturas_matriculas',
      joinColumn: { 
        name: 'idmatricula',
        referencedColumnName: 'id'
      },
      inverseJoinColumn: { 
        name: 'idasignatura',
        referencedColumnName: 'id',
      }
    })
    asignaturas: Asignatura[];

}

File: exam-activities/institutov2/src/main/java/es/iespuertodelacruz/nla/institutov2/entities/Matricula.java
package es.iespuertodelacruz.nla.institutov2.entities;

import java.io.Serializable;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import java.util.List;

/**
 * @author Nabil Leon Alvarez <@nalleon>
 */

/**
 * The persistent class for the matriculas database table.
 * 
 */
@Entity
@Table(name="matriculas")
@NamedQuery(name="Matricula.findAll", query="SELECT m FROM Matricula m")
public class Matricula implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(unique=true, nullable=false)
	private int id;

	@Column(name = "\"year\"")
	private int year;

	//bi-directional many-to-one association to Alumno
	@ManyToOne
	@JoinColumn(name="dni")
	@JsonIgnore
	private Alumno alumno;


	//bi-directional many-to-many association to Asignatura
	@ManyToMany(mappedBy="matriculas" /*, fetch=FetchType.LAZY*/  )
	private List<Asignatura> asignaturas;


	public Matricula() {
	}

	public int getId() {
		return this.id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public int getYear() {
		return this.year;
	}

	public void setYear(int year) {
		this.year = year;
	}

	public Alumno getAlumno() {
		return this.alumno;
	}

	public void setAlumno(Alumno alumno) {
		this.alumno = alumno;
	}

	public List<Asignatura> getAsignaturas() {
		return this.asignaturas;
	}

	public void setAsignaturas(List<Asignatura> asignaturas) {
		this.asignaturas = asignaturas;
	}

}